contracts/Libraries/Tupler.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.8.4;
+
+contract LibTupler {
+    function tupleSlicer(
+        uint256 offset,
+        uint256 size,
+        bytes memory slot
+    ) external returns (bytes memory) {
+        // the below should be equivalent to:
+        //        bytes memory ret = new bytes(size);
+        //        memcpy(slot, offset, ret, 0, size);
+        assembly {
+            mstore(add(slot, offset), size)
+            return(add(slot, offset), add(size, 32))
+        }
+    }
+
+    function memcpy(
```

Comment:
We should move this into a library that can be used in both places to avoid duplicating data.